nixos/modules/system/boot/systemd/dm-verity.nix

```diff
@@ -0,0 +1,57 @@
+{ config, lib, ... }:
+
+let
+  initrdCfg = config.boot.initrd.systemd.dmVerity;
+in
+{
+  options = {
+    boot.initrd.systemd.dmVerity = {
+      enable = lib.mkEnableOption "dm-verity" // {
+        description = ''
+          Mount verity-protected block devices in the initrd.
+
+          Enabling this option allows to use `systemd-veritysetup` and
+          `systemd-veritysetup-generator` in the initrd.
+        '';
+      };
+    };
+  };
+
+  config = lib.mkIf initrdCfg.enable {
+    assertions = [
+      {
+        assertion = initrdCfg.enable -> config.boot.initrd.systemd.enable;
+        message = ''
+          'boot.initrd.systemd.dmVerity.enable' requires 'boot.initrd.systemd.enable' to be enabled.
+        '';
+      }
+    ];
```

Comment:
```suggestion
  config = lib.mkIf (config.boot.initrd.systemd.enable && cfg.enable) {
```

Reply:
I think it's better to have the assertion here rather than silently not do anything, or what's your point for dropping it?

Reply:
Agree with @msanft here.